mastodon/src/main/java/org/joinmastodon/android/fragments/ComposeFragment.java

```diff
@@ -449,7 +449,7 @@ public void afterTextChanged(Editable s){
 				if(!mentions.contains(m))
 					mentions.add(m);
 			}
-			initialText=TextUtils.join(" ", mentions)+" ";
+			initialText=mentions.isEmpty() ? "" : TextUtils.join(" ", mentions)+" ";
```

Comment:
If the mentions string (containing e.g. "@Mastodon@mastodon.social") list is empty, use an empty string as the initial text, only otherwise join the mentions and add a single space after.